model/dependencies.go

```diff
@@ -127,15 +131,25 @@ func (di *dependencyIncluder) handle(pair TVPair, activationInfo *specificActiva
 	return true, nil
 }
 
-func (di *dependencyIncluder) updateDeactivationMap(activationInfo *specificActivationInfo, dep TVPair, generatedVariants []parserBV, pairSpecifiesActivation bool) {
-	if !variantExistsInGeneratedProject(generatedVariants, dep.Variant) {
+func (di *dependencyIncluder) updateDeactivationMap(pair TVPair, generatedVariants []parserBV, pairSpecifiesActivation bool) {
+	if !variantExistsInGeneratedProject(generatedVariants, pair.Variant) {
 		// If the dependency has not yet been added to deactivateGeneratedDeps, or if the
 		// original pair needs to be active, we update deactivateGeneratedDeps.
 		// We ultimately will only deactivate new dependencies where deactivateGeneratedDeps[pair] = true.
 		// If deactivateGeneratedDeps[pair] = false it signifies that there was at least
 		// one pair that depends on this new dep being active - so we cannot deactivate it.
-		if _, ok := di.deactivateGeneratedDeps[dep]; !ok || !pairSpecifiesActivation {
-			di.deactivateGeneratedDeps[dep] = pairSpecifiesActivation
+		if _, foundPair := di.deactivateGeneratedDeps[pair]; !foundPair || !pairSpecifiesActivation {
+			di.deactivateGeneratedDeps[pair] = pairSpecifiesActivation
+			bvt := di.Project.FindTaskForVariant(pair.TaskName, pair.Variant)
+			// Also need to deactivate the dependencies of the newly deactivated tasks
+			if bvt != nil {
+				deps := di.expandDependencies(pair, bvt.DependsOn)
```

Comment:
Can you remind me why we don't need to recurse on this?  We may also want to comment aggressively because we're not gonna remember how this works in a month i think haha

Reply:
We do, thanks for the catch. `expandDependencies` only finds dependencies at the first level which I missed.  Updated this to recursion and added another unit test case to confirm that the full dependency tree gets its deactivation value updated.